### PR TITLE
Use go 1.12 for tests and deprecate go 1.9

### PR DESCRIPTION
**Pull request by techknowlogick**
Blocked until 1.12 docker images are released

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ pipeline:
       event: [ push, tag, pull_request ]
 
   build-without-gcc:
-    image: golang:1.9
+    image: golang:1.10
```
Comment (kolaente) [inline .drone.yml]: Shouldn't this also be `1.12`?

Comment (techknowlogick) [reply .drone.yml]: this step is kept as the lowest version of golang that we support. Currently we support all the way down to 1.9 but this changes it to be 1.10 be the oldest we support.

**Comment by techknowlogick** [comment]
Waiting on https://github.com/docker-library/golang/pull/264 to be merged

**Review by sapk — APPROVED** [review]
LGTM just need to wait for the docker image.

Just to be sure I checked that you are also ready to update the xgo image :smile: 
https://github.com/techknowlogick/xgo/pull/2

**Comment by techknowlogick** [comment]
@sapk yes, I will merge the xgo PR after we release 1.7.3, but before 1.8.0. Thanks for review 😃

**Comment by techknowlogick** [comment]
Now awaiting https://github.com/docker-library/official-images/pull/5473

**Comment by techknowlogick** [comment]
No longer blocked. golang v1.12 image released.

I've also added updates to remove some duplicated step names, as this would block a move to drone 1.0.0 if not done.

**Review by sapk — APPROVED** [review]

**Comment by zeripath** [comment]
Make lg-tm work

**Review by zeripath — APPROVED** [review]

**Review by appleboy — APPROVED** [review]

**Pull request merged**